Source/Core/Duality/Utility/Coroutines/WaitUntil.cs

```diff
@@ -50,6 +60,10 @@ internal void Update()
 				case WaitType.RealTime:
 					this.internalValue -= Time.UnscaledDeltaTime;
 					break;
+
+				case WaitType.Coroutine:
+					this.internalValue = this.subroutineMoveNext() ? float.PositiveInfinity : 0;
```

Comment:
Since the func will return a int now you can just assign the result to internalValue